std/special/compiler_rt/truncXfYf2.zig

```diff
@@ -79,8 +85,8 @@ inline fn truncXfYf2(comptime dst_t: type, comptime src_t: type, a: src_t) dst_t
         // a underflows on conversion to the destination type or is an exact
         // zero.  The result may be a denormal or zero.  Extract the exponent
         // to get the shift amount for the denormalization.
-        const aExp: u32 = aAbs >> srcSigBits;
-        const shift: u32 = srcExpBias - dstExpBias - aExp + 1;
+        const aExp = aAbs >> srcSigBits;
+        const shift = srcExpBias - dstExpBias - aExp + 1;
```

Comment:
Suggestion: `@truncate(u32, ...)`?  The exponent and the shift fit in 32 bits.

I put in the u32 intentionally because I figured that for performance reasons you don't want to use a 64 or 128 bits type when you can avoid it.

Reply:
I can add an `@intCast(u32, ...)` here. I omitted it here for the moment since the narrowing doesn't work implicitly for the longer `src_t` types (such as `f128`).